straxen/plugins/led_calibration.py

```diff
@@ -0,0 +1,114 @@
+import numpy as np
+import pandas as pd
+from tqdm import tqdm
+import numba
+from numba import njit
+
+#export, __all__ = strax.exporter()
+# QUESTION: what does 'export, __all__ = strax.exporter()' do?
+# ANSWER: [fill in]
+
+@strax.takes_config(
+    strax.Option('LED_window',
+                 default=(125, 250),
+                 help="Window (samples) where we expect the signal in LED calibration"),
+    strax.Option('noise_window',
+                 default=(350, 475),
+                 help="Window (samples) to analysis the noise"),
+    strax.Option('light_level',
+                 default=(0,248),
+                 help="Three different light level for XENON1T: (0,36), (37,126), (127,248). Defalt value: all the PMTs")
+)
+
+
+class LEDCalibration(strax.Plugin):
+    
+    '''
+    Preliminary version, several parameters to set during commisioning.
+    LEDCalibration returns: channel, time, dt, lenght, Area, amplitudeLED and amplitudeNOISE.
+    The new variables are:
+    - Area: Area computed in the given window, averaged over 6 windows that have the same starting sample and different end samples.
+    - amplitudeLED: peak amplitude of the LED on run in the given window.
+    - amplitudeNOISE: amplitude of the LED on run in a window far from the signal one.
+    '''
+    
+    __version__ = '0.0.8'
+    print(__version__)
+    depends_on = ('raw_records',)
+    # Options below copied from other plugins, need to be reviewed by an expert
+    data_kind = 'led_cal_0' 
+    compressor = 'zstd'
+    parallel = 'process'
+    rechunk_on_save = False
+    
+    dtype = [('area', np.int32, 'Area averaged in integration windows'),
+             ('amplitudeLED', np.int32, 'Amplitude in LED window'),
+             ('amplitudeNOISE', np.int32, 'Amplitude in off LED window'),
+             ('channel', np.int16, 'Channel'),
+             ('time', np.int64, 'Start time of the interval (ns since unix epoch)'),
+             ('dt', np.int16, 'Time resolution in ns'),
+             ('length', np.int32, 'Length of the interval in samples')
+            ]
+    
+    def compute(self, raw_records):
+        
+        r = raw_records[(raw_records['channel'] >= self.config['light_level'][0])&(raw_records['channel'] <= self.config['light_level'][1])]
+        # TODO: to change during nT commissioning or add in configuration options
+        temp = np.zeros(len(r), dtype=self.dtype)
+        
+        temp['channel'] = r['channel']
+        temp['time'] = r['time']
+        temp['dt'] = r['dt']
+        temp['length'] = r['length']
+        
+        on, off = get_amplitude(r, self.config['LED_window'], self.config['noise_window'])
+        temp['amplitudeLED'] = on['amplitudeLED']
+        temp['amplitudeNOISE'] = off['amplitudeNOISE']
+
+        area = get_area(r, self.config['LED_window'])
+        temp['area'] = area['area']
+
+        
+        return temp
+
+# QUESTIONS: can some nice functions of numba.njit be used? What does @export mean?
```

Comment:
Maybe you could do a speed test comparing this function using numpy and using njit like in this notebook [here](https://github.com/XENONnT/workshop_strax_chicago/blob/master/projects/LED_calibration/LEDcal_evan.ipynb) (In [106])

Reply:
I can try it

Reply:
TypingError: Failed in nopython mode pipeline (step: nopython frontend)
Invalid use of Function(<built-in function zeros>) with argument(s) of type(s): (int64, dtype=list(tuple(unicode_type x 2)))
 * parameterized
In definition 0:
    All templates rejected with literals.
In definition 1:
    All templates rejected without literals.
This error is usually caused by passing an argument of a type that is unsupported by the named function.
[1] During: resolving callee type: Function(<built-in function zeros>)
[2] During: typing of call at /home/gvolta/XENONnT/straxen/straxen/plugins/led_calibration.py (88)


File "../straxen/straxen/plugins/led_calibration.py", line 88:
def get_amplitude(raw_records, led_window, noise_window):
    <source elided>
    '''
    on = np.zeros((len(raw_records)), dtype=[('channel','int16'),('amplitude_led', '<i4')])
    ^

This is not usually a problem with Numba itself but instead often caused by
the use of unsupported features or an issue in resolving types.

To see Python/NumPy features supported by the latest release of Numba visit:
http://numba.pydata.org/numba-doc/dev/reference/pysupported.html
and
http://numba.pydata.org/numba-doc/dev/reference/numpysupported.html

For more information about typing errors and how to debug them visit:
http://numba.pydata.org/numba-doc/latest/user/troubleshoot.html#my-code-doesn-t-compile

If you think your code should work with Numba, please report the error message
and traceback, along with a minimal reproducer at:
https://github.com/numba/numba/issues/new

I got this error after adding @njit. Am I using in the wrong way?